catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -207,13 +216,43 @@ private void fetchNextResults() {
       try {
         SourceResponse response = queryFunction.query(queryRequestCopy);
 
-        if (response.getResults().size() == 0) {
+        final List<Result> resultList = response.getResults();
+
+        // Because some of the results may be filtered out by the catalog framework's
+        // plugins, we need a way to know the actual page size and increment currentIndex based
+        // on that number instead of using the result list size.
+        // If the property is not present, we will have no option but to fallback to the size
+        // of the (potentially filtered) resultList.
+        //
+        // This means that if the filtered results size is zero, but the raw number of results
+        // had been greater than zero, we will not find results beyond the filtered gap. In practice
+        // this should not happen, as queries will run through the QueryOperations.query() method;
```

Comment:
Wrong again, Kiddo.

Reply:
I'm REALLY NOT HAPPY that SonarQube 
a) has any opinion whatsoever about inline comments where are a valid and useful tool to explicate complex or otherwise confusing code
b) is potentially choking on my perfectly correct use of a semi-colon to mark out an independent clause.

Would SonarQube rather I used a grammatically incorrect _comma splice_?!?

Joking aside, if there's a way to tell SonarQube to NEVER EVER EVER comment about the existence of an inline comment, we need to do that.